### PR TITLE
[docs] Fix wrong required properties for HDFS2 sink

### PR DESCRIPTION
**Pull request by BewareMyPower**
### Motivation

From `HdfsSinkConfig.validate` method in `org.apache.pulsar.io.hdfs2.sink`, `fileExtension` (if `compression` is None) and `filenamePrefix` are required, but the documents marked them as not required.

### Modifications

Fix the *required* properties and modify example configs.

### CODE REVIEW

File: site2/docs/io-hdfs2-sink.md [hunk 1]
```diff
@@ -21,8 +21,8 @@ The configuration of the HDFS2 sink connector has the following properties.
 | `compression` | Compression |false |None |The compression code used to compress or de-compress the files on HDFS. <br/><br/>Below are the available options:<br/><li>BZIP2<br/><li>DEFLATE<br/><li>GZIP<br/><li>LZ4<br/><li>SNAPPY|
 | `kerberosUserPrincipal` |String| false| None|The principal account of Kerberos user used for authentication. |
 | `keytab` | String|false|None| The full pathname of the Kerberos keytab file used for authentication. |
-| `filenamePrefix` |String| false |None |The prefix of the files created inside the HDFS directory.<br/><br/>**Example**<br/> The value of topicA result in files named topicA-. |
-| `fileExtension` | String| false | None| The extension added to the files written to HDFS.<br/><br/>**Example**<br/>'.txt'<br/> '.seq' |
+| `filenamePrefix` |String| true if `compression` is None | None |The prefix of the files created inside the HDFS directory.<br/><br/>**Example**<br/> The value of topicA result in files named topicA-. |
```
Comment (Huanli-Meng) [inline site2/docs/io-hdfs2-sink.md:24]:
```suggestion
| `filenamePrefix` |String| true, if `compression` is set to `None`. | None |The prefix of the files created inside the HDFS directory.<br/><br/>**Example**<br/> The value of topicA result in files named topicA-. |
```

**Comment by Huanli-Meng** [comment]
@Jennifer88huang , it seems this is suitable for previous releases. Please cherry-pick to release 2.6.1.

File: site2/docs/io-hdfs2-sink.md (hunk 1 above)
Comment (BewareMyPower) [reply site2/docs/io-hdfs2-sink.md:24]: OK, I'll fix it.

**Comment by Jennifer88huang** [comment]
@Huanli-Meng Got it, thanks for your reminder.

**Pull request merged**